### ISSUE TITLE
Disk full when vdi clone

### ISSUE THREAD

**Issue by tankitty** [sheepdog/sheepdog#237]
sheep disk full when vdi clone  (3copy)
i have 18 nodes sometime 1 node Use > 95%
can't clone image because disk space not enough (1 node)

e.g. node 0-16 is 60-70% used node17 is 95% used

any idea ?

Id	Size	Used	Avail	Use%
 0	3.1 TB	1.9 TB	1.2 TB	 60%
 1	3.1 TB	1.9 TB	1.2 TB	 62%
 2	3.1 TB	1.9 TB	1.1 TB	 63%
 3	1.3 TB	937 GB	400 GB	 70%
 4	1.3 TB	967 GB	374 GB	 72%
 5	447 GB	335 GB	112 GB	 74%
 6	223 GB	178 GB	45 GB	 80%
 7	223 GB	174 GB	49 GB	 78%
 8	223 GB	165 GB	59 GB	 73%
 9	223 GB	177 GB	46 GB	 79%
10	223 GB	160 GB	63 GB	 71%
11	447 GB	326 GB	121 GB	 72%
12	447 GB	339 GB	108 GB	 75%
13	223 GB	185 GB	38 GB	 82%
14	1.3 TB	1.0 TB	314 GB	 76%
15	119 GB	89 GB	30 GB	 74%
16	112 GB	90 GB	22 GB	 80%
17	119 GB	84 GB	35 GB	 70%
Total	16 TB	11 TB	5.3 TB	 67%

Total virtual image size	3.5 TB

**Comment by mitake** [comment]
@tankitty vdi cloning just introduces a new inode object that consumes only 12MB, so it is strange. Could you provide full command line for cloning operation?

**Comment by tankitty** [comment]
dog vdi clone -n -s v1 src-img dst-img

**Comment by mitake** [comment]
You are using `-n` option that causes deep copy of objects. It consumes bunch of disk space. Is it really needed for your usecase?

**Comment by tankitty** [comment]
i have many virtual machine, i need to copy os image to new vdi if without -n image will share from master image ?

**Comment by mitake** [comment]
Without `-n`, cloned VDIs don't consume disk space until the shared objects are overwritten.

**Comment by tankitty** [comment]
can i share more than 1000 VMs ?

**Comment by mitake** [comment]
Yes, a number of cloned VDIs doesn't matter for the cloning functionality.

**Comment by tankitty** [comment]
you'll recommend use clone from a snapshot ?

**Comment by mitake** [comment]
Yes, creating a base image snapshot -> clone without `-n` would be the best practice for your case.

**Comment by tankitty** [comment]
ok, thanks : )

**Comment by tankitty** [comment]
Current sheepdog traffic on my localnetwork

<img width="1092" alt="screen shot 2559-04-28 at 2 24 10 pm" src="https://cloud.githubusercontent.com/assets/15312479/14878304/e9986d2e-0d4c-11e6-9ed5-721ddbdf2783.png">
<img width="1136" alt="screen shot 2559-04-28 at 2 23 57 pm" src="https://cloud.githubusercontent.com/assets/15312479/14878306/eaaafac4-0d4c-11e6-9289-f940fd14ab19.png">
<img width="1115" alt="screen shot 2559-04-28 at 2 24 16 pm" src="https://cloud.githubusercontent.com/assets/15312479/14878308/ebb3f056-0d4c-11e6-8b86-11428bcb7c04.png">

**Comment by mitake** [comment]
If your network has 4Gbps bandwidth, it is not saturated yet (I'm not sure the above traffic patterns include the cloning operation). Maybe cloning without `-n` works well for your purpose. Could you try it?

**Comment by tankitty** [comment]
if we have 17 nodes with 3 copy when i clone from snapshot without -n the master object will shared only from 3 dog node right?

**Comment by mitake** [comment]
The objects will be stored only in 3 sheep nodes.

**Comment by tankitty** [comment]
OK , this is traffic pattern when i create 3 vm one by one ( dog vdi clone with -n option ) 40GB image

start 3:00 am

![screen shot 2559-05-01 at 3 15 37 am](https://cloud.githubusercontent.com/assets/15312479/14938401/6dd8ed6a-0f4c-11e6-9877-a37cb6d18793.png)
![screen shot 2559-05-01 at 3 15 56 am](https://cloud.githubusercontent.com/assets/15312479/14938402/6ddb614e-0f4c-11e6-9dca-6dfa7bcbc081.png)
![screen shot 2559-05-01 at 3 16 03 am](https://cloud.githubusercontent.com/assets/15312479/14938403/6ddc1e22-0f4c-11e6-8b50-7d94b5fce0ee.png)

**Comment by mitake** [comment]
Your network seems not to be saturated. Could you try cloning without `-n` and do some benchmarking? I'm interested in the result of such workload (and it would work well, I think).
